src/snippets/dto/create-snippet.dto.ts

```diff
@@ -1,6 +1,18 @@
 import { ApiProperty } from '@nestjs/swagger';
+import { IsBoolean, IsOptional, IsString } from 'class-validator';
+import { RequiredWhenTrue } from '../../decorators/RequiredWhenTrue';
 
 export class CreateSnippetDto {
   @ApiProperty()
+  @IsString()
   content: string;
+
+  @ApiProperty()
+  @IsBoolean()
+  @IsOptional()
+  isProtected?: boolean;
```

Comment:
hmmm, czemu FE miałby wysyłać to pole jako część payload'u? Czy sam fakt wysłania hasła obok payload'u nam nie wystarczy?